src/XMakeCommandLine/UnitTests/XMake_Tests.cs

```diff
@@ -633,7 +632,8 @@ public void ConfigurationInvalid()
                     "</Project>";
                 File.WriteAllText(pathToProjectFile, projectString);
 
-                output = RunProcessAndGetOutput(newPathToMSBuildExe, "\"" + pathToProjectFile + "\"", expectSuccess: false);
+                var msbuildParameters = "\"" + pathToProjectFile + "\"";
+                output = runMSBuildWithinCoreRun(newPathToMSBuildExe, msbuildParameters, expectSuccess: false);
```

Comment:
Agreed. How about rename RunMSBuildWithinCoreRun to just ExecMSBuild and have it do the right thing there based on if we need corerun or not.